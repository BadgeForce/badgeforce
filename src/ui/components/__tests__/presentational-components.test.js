import React from "react";
import renderer from "react-test-renderer";
import { shallow } from "enzyme";
import CSVErrors from "../csv-errors";
import { csvErrors } from "./fixtures/csv";
import Intro from "../intro";
import CSVArea from "../csv";

describe("Presentational component tests", () => {
    describe("CSVErrors component tests", () => {
        it("should render proper CSVErrors component with no errors", () => {
            const tree = renderer.create(<CSVErrors errors={[]} />).toJSON();
            expect(tree).toMatchSnapshot();
        });
        it("should render proper CSVErrors component with more than one error", () => {
            const tree = renderer.create(<CSVErrors errors={csvErrors} />).toJSON();
            expect(tree).toMatchSnapshot();
        });

        describe("CSVErrors visibility tests", () => {
            let csvErrorsComp;
            beforeEach(() => {
                csvErrorsComp = shallow(<CSVErrors errors={csvErrors} />);
            });

            it("should collapse and uncollapse CSVErrors area", () => {
                csvErrorsComp.find("#csv-errors-visibility-button").simulate("click");
                expect(csvErrorsComp.state("open")).toEqual(false);
                expect(csvErrorsComp).toMatchSnapshot();

                csvErrorsComp.find("#csv-errors-visibility-button").simulate("click");
                expect(csvErrorsComp.state("open")).toEqual(true);
            });
        });
    });

    describe("Intro component tests", () => {
        it("should render proper Intro component", () => {
            const tree = renderer.create(
                <Intro />
            ).toJSON();
            expect(tree).toMatchSnapshot();
        });
    });

    describe("CSVArea component tests", () => {
        it("should render proper CSVArea component", () => {
            const tree = renderer.create(
                <CSVArea />
            ).toJSON();
            expect(tree).toMatchSnapshot();
        });
    });
});
