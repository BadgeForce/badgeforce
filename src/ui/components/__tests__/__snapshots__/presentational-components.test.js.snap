// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Presentational component tests CSVArea component tests should render proper CSVArea component 1`] = `
<div
  className="span10"
>
  <br />
  <h3>
    Select a CSV file with account specific data.
  </h3>
  <div>
    <input
      accept="text/csv"
      onChange={[Function]}
      type="file"
    />
  </div>
  <br />
</div>
`;

exports[`Presentational component tests CSVErrors component tests CSVErrors visibility tests should collapse and uncollapse CSVErrors area 1`] = `
<div
  className="span10"
>
  <br />
  <h3
    style={
      Object {
        "color": "red",
      }
    }
  >
    Errors in your CSV file.
  </h3>
  <div>
    <button
      className="btn"
      id="csv-errors-visibility-button"
      onClick={[Function]}
    >
      Open Errors List
    </button>
  </div>
  <br />
</div>
`;

exports[`Presentational component tests CSVErrors component tests should render proper CSVErrors component with more than one error 1`] = `
<div
  className="span10"
>
  <br />
  <h3
    style={
      Object {
        "color": "red",
      }
    }
  >
    Errors in your CSV file.
  </h3>
  <div>
    <button
      className="btn"
      id="csv-errors-visibility-button"
      onClick={[Function]}
    >
      Collapse Errors List
    </button>
    <div>
      <div>
        <h4>
          Row: 
          1
        </h4>
        <ul>
          <li>
            SalesForce Account Id cannot be left empty, must start with 001 and be followed by only letters and numbers
          </li>
          <li>
            Primary Phone cannot be left empty and must be a valid phone number
          </li>
        </ul>
      </div>
      <div>
        <h4>
          Row: 
          2
        </h4>
        <ul>
          <li>
            SalesForce Account Id cannot be left empty, must start with 001 and be followed by only letters and numbers
          </li>
        </ul>
      </div>
    </div>
  </div>
  <br />
</div>
`;

exports[`Presentational component tests CSVErrors component tests should render proper CSVErrors component with no errors 1`] = `
<div
  className="span10"
>
  <br />
  <h3
    style={
      Object {
        "color": "red",
      }
    }
  >
    Errors in your CSV file.
  </h3>
  <div>
    <button
      className="btn"
      id="csv-errors-visibility-button"
      onClick={[Function]}
    >
      Open Errors List
    </button>
    <div />
  </div>
  <br />
</div>
`;

exports[`Presentational component tests Intro component tests should render proper Intro component 1`] = `
<div
  className="span10"
>
  <br />
  <h1>
    Global Account Registration for YodleLive
  </h1>
</div>
`;
